site/en/docs/extensions/mv3/geolocation/index.md

```diff
@@ -0,0 +1,204 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Using geolocation"
+seoTitle: "How to use geolocation in Chrome Extensions"
+date: 2023-06-20
+description: How to use geolocation in Chrome Extensions service worker, popup, sidepanel or content script
+---
+
+If you want to get geolocation information in your Chrome extension, use the same [`navigator.geolocation`](geolocation) DOM API that any web site normally would. The reason this article exists is because Chrome extensions handle permission to sensitive data differently than normal websites. Geolocation is very sensitive data, and as such browsers take effort to make sure that people using browsers are fully aware and in control of when and where their exact location is shared.
+
+
+## Using geolocation in MV3 extensions
+
+On the web, browsers safeguard user's geolocation data by showing them a prompt, asking the user to grant that specific origin access to their location. The same permission model is not always appropriate for extensions.
+
+
+{% Img src="image/DXqUldooyJOUnj3qXSYLHbUgUI93/nYRDFWbb280yEp90XDTI.png", alt="A screenshot of the permission prompt you see when a website requests access to the geolocation api", width="800", height="419" %}
+
+Permissions are not the only difference. As mentioned above, `navigator.geolocation` is a \_\_DOM\_\_ API, that is, something that is a part of the APIs that make up web sites. As a result, it is not accessible inside Worker contexts, like the [extension service worker](sw) that is the backbone of manifest v3 extensions. You can absolutely still use `geolocation`, though. There are just nuances with how and where you use it.
+
+
+### Using geolocation in Service Workers
+
+There is no `navigator` object inside of service workers. It is only available inside of contexts that have access to a page's `document` object. To get access inside of a service worker, use an [`Offscreen Document`](offscreen). This is a feature of extensions that gives you access to an html file you can bundle with our extension. You can read more about Offscreen Documents in the [documentation](offscreen). 
+
+To get started add `offscreen` to the `permissions` section of our manifest.
+
+{% Label %}manifest.json:{% endLabel %}
+
+```js
+{
+  "name": "My extension",
+    ...
+  "permissions": [
+    ...
+   "offscreen"
+  ],
+  ...
+}
+```
+
+After adding the `offscreen` permission, add an html file to our extension that will include your offscreen document. In this case, we are not using any of the content of the page, and so this can be a nearly blank file, it just needs to be a smal, html file that loads in our script.
```

Comment:
```suggestion
After adding the `offscreen` permission, add an html file to your extension that includes your offscreen document. This case isn't using any of the content of the page, and so this can be a nearly blank file. It just needs to be a small, html file that loads in your script.
```